crates/host/src/actor/mod.rs

```diff
@@ -0,0 +1,291 @@
+        match self {
+            Self::Module(module) => Self::Module(module.logging(logging)),
+            Self::Component(component) => Self::Component(component.logging(logging)),
+        }
+    }
+
+    /// Configure actor to inherit standard output of the process
+    #[must_use]
+    pub fn inherit_stdout(self) -> Self {
+        match self {
+            Self::Module(module) => Self::Module(module.inherit_stdout()),
+            Self::Component(component) => Self::Component(component.inherit_stdout()),
+        }
+    }
+
+    /// Configure actor to inherit standard error of the process
+    #[must_use]
+    pub fn inherit_stderr(self) -> Self {
+        match self {
+            Self::Module(module) => Self::Module(module.inherit_stderr()),
+            Self::Component(component) => Self::Component(component.inherit_stderr()),
+        }
+    }
+
+    /// Instantiate the configured actor
+    ///
+    /// # Errors
+    ///
+    /// Fails if the underlying [Component::instantiate] or [Module::instantiate]
+    #[instrument]
+    pub async fn instantiate(self) -> anyhow::Result<Instance> {
+        match self {
+            Self::Module(module) => module.instantiate().await.map(Instance::Module),
+            Self::Component(component) => component.instantiate().await.map(Instance::Component),
+        }
+    }
+
+    /// Instantiate the configured actor and invoke an operation on it.
+    ///
+    /// # Errors
+    ///
+    /// Fails if the underlying [Component::call] or [Module::call]
+    #[instrument(skip(operation, payload))]
+    pub async fn call(
+        self,
+        operation: impl AsRef<str>,
+        payload: Option<impl Into<Vec<u8>> + AsRef<[u8]>>,
+    ) -> anyhow::Result<Result<Option<Vec<u8>>, String>> {
+        let operation = operation.as_ref();
+        self.instantiate()
+            .await
+            .context("failed to instantiate actor")?
+            .call(operation, payload)
+            .await
+            .with_context(|| format!("failed to call operation `{operation}` on actor"))
+    }
+}
+
+/// A pre-loaded, configured wasmCloud actor instance, which is either a module or a component
+#[derive(Debug)]
+pub enum Instance {
+    /// WebAssembly module containing an actor
+    Module(ModuleInstance),
+    /// WebAssembly component containing an actor
+    Component(ComponentInstance),
+}
+
+impl Instance {
+    /// Invoke an operation on an [Instance] producing a response
+    ///
+    /// # Errors
+    ///
+    /// Outermost error represents a failure in calling the actor, innermost - the
+    /// application-layer error originating from within the actor itself
+    #[instrument(skip_all)]
+    pub async fn call(
+        &mut self,
+        operation: impl AsRef<str>,
+        payload: Option<impl Into<Vec<u8>> + AsRef<[u8]>>,
+    ) -> anyhow::Result<Result<Option<Vec<u8>>, String>> {
+        let operation = operation.as_ref();
```

Comment:
same comment here re: the `as_ref` call

Reply:
No, because we also want to be able to print it on failures